### PR TITLE
Replace the address in SignTypedData_v4 signatures with a 'Verify contract details' link 

### LINKED ISSUE #15912: Replace the address being displayed in SignTypedData_v4 signatures "title" with a 'Verify Contract Details' link
## Resources
[Figma design file](figma.com/file/U2Ojf3cJP4PvIYRiRfUya0/Improve-Signatures-(v1)?node-id=1960%3A806)
The scope of this ticket is very similar to the scope of https://github.com/MetaMask/metamask-extension/issues/15697

## Acceptance Criteria
1. When signTypedData_v4 confirmation screen is displayed it now does not show an address below `Signature Request` title;
2. When signTypedData_v4 confirmation screen is displayed it now shows a `Verify contract details` link as in design (it should display `Verify contract details` instead of `Verify message details` as currently is in the design and ignore the header form the screenshot as it i snot the current one);
3. Clicking on this link will take you to this new contract detail modal that was create on https://github.com/MetaMask/metamask-extension/issues/15226 (It's the same one used on the new token and NFT allowance confirmation screens);
4. Modal should display only one contract detail (on the other confirmation screens it usually displays 2 different contract info);
5. Use the design pattern from the third modal and the wording as well - Title: "Contract details" and wording: "To protect yourself against scammers, take a moment to verify contract details. 
6. Since there is no URL do not add the "Learn how to verify contract details." sentence
7. Instead of reading `Contract requesting spending cap`, it should read `Contract requesting signature`;
8. The contract displayed is the address being passed in the signature message domain ([MM signature docs](https://docs.metamask.io/guide/signing-data.html#sign-typed-data-v4));
9. This modal has only one button with copy `Got it` as in the third screenshot below.Screen Shot 2022-09-20 at 22 57 55
Screen Shot 2022-09-20 at 22 57 45
Screen Shot 2022-09-20 at 22 58 24

## Steps to Reproduce
1. Open the test dapp https://metamask.github.io/test-dapp/
2. Under the Sign Typed Data V4 section, click on "Sign"

### PR DESCRIPTION
**Pull request by filipsekulic**
## Explanation
Replace the address being displayed in SignTypedData_v4 signatures 'title' with a 'Verify contract details' link.

## More Information
* Fixes #15912 

### Before

https://user-images.githubusercontent.com/92310504/195531386-101c6fd5-05e7-49db-8dbb-25369dff881f.mov

### After

https://user-images.githubusercontent.com/92310504/195606622-6e6cc85d-5055-4f2f-affd-7ac34773585e.mov

## Manual Testing Steps

1. Open the Test Dapp: https://metamask.github.io/test-dapp/
2. Under the section `Sign Typed Data V4` click on `SIGN`
3. Then click on a `Verify contract details` and a modal will show up

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
Verified by QA

**Review by darkwing — DISMISSED** [review]
This looks great!

**Review by brad-decker — DISMISSED** [review]

File: test/e2e/tests/signature-request.spec.js [hunk 1]
```diff
@@ -132,19 +132,19 @@ describe('Sign Typed Data V3 Signature Request', function () {
...
         assert.equal(await origin.getText(), 'http://127.0.0.1:8080');
         assert.equal(
-          await address.getText(),
-          `${publicAddress.slice(0, 8)}...${publicAddress.slice(
-            publicAddress.length - 8,
-          )}`,
+          await verifyContractDetailsButton.getText(),
+          'Verify contract details',
         );
         assert.equal(await messages[4].getText(), 'Hello, Bob!');
```
[the remark this answers is not in the document]
Comment (bschorchit) [reply test/e2e/tests/signature-request.spec.js:154]: Could you give this a re-review when you have the chance, @jpuri? Thank you <3

**Review by brad-decker — APPROVED** [review]

**Review by darkwing — APPROVED** [review]
Great work @filipsekulic !

**Pull request merged**